### PR TITLE
Favorite event

### PR DESCRIPTION
**Pull request by asogwachiuba**
What is in the PR:
- Refactored the favoriting event logic all throught the app
- Changed the favoriting api to liking api as likes represents favorites currently on events.
- Updated the profile favoriting feature to match the changes
- No changes were made to the UI

### CODE REVIEW

File: lib/core/services/event/event_service.dart [hunk 1]
```diff
@@ -53,6 +54,8 @@ abstract class EventServices {
   Future<ApiResult<MoreEventsResponse>> getMoreEvents({
     required String eventId,
   });
+
+  Future<ApiResult<List<EventModel>>> getLikedEvent();
```
Comment (divincoder) [inline lib/core/services/event/event_service.dart:58]: rename to getLikedEvents

File: lib/features/events/views/event_detail/event_detail_viewmodel.dart [hunk 2]
```diff
@@ -167,12 +167,8 @@ class EventDetailViewModel extends IbloovViewModel {
...
 
   /// Checks if event is the user's favorite
-  isEventFavorited() {
-    logger.d("Checking if its a favorite");
-    String userId = _databaseService.getCurrentUser()?.id ?? "";
-    logger.d(userId);
-    bool isFavorite_ = currentEvent?.favoritedBy?.contains(userId) ?? false;
-    logger.d("This event is favourited? $isFavorite_");
+  isEventLiked() {
+    bool isFavorite_ = currentEvent?.userLiked ?? false;
```
Comment (divincoder) [inline lib/features/events/views/event_detail/event_detail_viewmodel.dart:171]: why did you change to this? Was this advised by mitchel? Because that liked doesn't work before why we are using favorited by

Comment (asogwachiuba) [reply lib/features/events/views/event_detail/event_detail_viewmodel.dart:171]: Every change here was discussed with Mithcel as Favoriting on event is one sided(i.e can't be unfavorited) and like is what is used on the web for favoriting. Also like api is a toggle so it provides the required solution

File: lib/core/services/event/event_service.dart (hunk 1 above)
Comment (asogwachiuba) [reply lib/core/services/event/event_service.dart]: Done

File: lib/features/events/views/event_detail/event_detail_viewmodel.dart (hunk 2 above)
Comment (divincoder) [reply lib/features/events/views/event_detail/event_detail_viewmodel.dart:190]: ok

File: lib/core/services/event/event_service.dart [hunk 3]
```diff
@@ -42,7 +43,7 @@ abstract class EventServices {
     bool? isPaid,
   });
 
-  Future<ApiResult<bool>> toggleFavoriteForEvent({
+  Future<ApiResult<bool>> toggleEventLike({
```
[the remark this answers is not in the document]
Comment (divincoder) [reply lib/core/services/event/event_service.dart]: alright, liked the old one better

**Review by divincoder — APPROVED** [review]
